index.js

```diff
@@ -534,13 +534,21 @@ export const matchStatus = f => fs => res => {
   return (hasProp (statusCode) (fs) ? fs[statusCode] : f) (res);
 };
 
-// mergeUrls :: Url -> Any -> String
-const mergeUrls = base => input => (
+// mergeUrls :: (Url, Any) -> String
+const mergeUrls = (base, input) => (
   typeof input === 'string' ?
   new URL (input, base).href :
   base
 );
 
+// sameHost :: (Url, Url) -> Boolean
+const sameHost = (parent, child) => {
+  const p = new URL (parent);
+  const c = new URL (child);
+  return p.host === c.host ||
+         c.host.endsWith (p.host) && c.host[p.host.length] === '.';
+};
+
```

Comment:
I think that the subdomain check is not correct, might just work with the test case because `sub` has a length of 3, as `com` does.

```node
> sameHost('http://domain.com', 'http://mega-sub.domain.com')
false
> sameHost('http://domain.com', 'http://sub.domain.com')
true
```

<hr/>

If `c.host` is `mega-sub.domain.com`, and `p.host` is `domain.com`. Then `p.host.length` is `10` and `c.host[p.host.length]` is `o`. However, `c` is a subdomain of `p`.

Shouldn't this last check be;

```javascript
  return p.host === c.host || c.host.endsWith ('.' + p.host);
```

Reply:
Wow, good catch. You are completely right. I don't know what I was thinking when writing this piece of code, but looking back at it it's obvious that it's wrong.

Reply:
Addressed in https://github.com/fluture-js/fluture-node/pull/13/commits/125e4474f910c1507f8ec3232848626fbc0f55c4 [[diff](https://github.com/fluture-js/fluture-node/compare/73832d943ecda0c5b42b93afb4583c252b82a803..125e4474f910c1507f8ec3232848626fbc0f55c4)]